cxx4opencl/address_spaces.txt

```diff
@@ -20,52 +20,25 @@ logic.
 
 ==== Casts
 
-C-style casts follow `OpenCL C v2.0 rules (s6.5.5)`. All cast operators
-permit conversion to generic address space implicitly. However converting
-from generic to named address spaces can only be done using
-`addrspace_cast`. Note that conversions between `__constant` and any other
-other address space are disallowed.
-
-TODO: Example with `addrspace_cast`.
-
-[[addrspace-deduction]]
-==== Deduction
-
-Address spaces are not deduced for:
-
-  * non-pointer/non-reference template parameters or any dependent types
-    except for template specializations.
-  * non-pointer/non-reference class members except for static data members
-    that are deduced to `__global` address space.
-  * non-pointer/non-reference alias declarations.
-  * decltype expressions.
+C-style casts follow rules of `OpenCL C v2.0 s6.5.5`. Conversions to generic
+address space can be made by any {cpp} cast operator (as an implicit
+conversion); converting from generic to named address space can only be done
+using dedicated `addrspace_cast` operator. Operator `addrspace_cast` can only
+convert between address spaces and no other conversions are allowed to occur.
+Note that conversions between `__constant` and any other other address space
+are disallowed.
 
 [source,c]
```

Comment:
I've never needed to try before so I don't know if it works.  But is `[source,cpp]` or similar supported, and does it result in any improvements?

Reply:
Works nicely! Thanks!

Reply:
Changed in https://github.com/KhronosGroup/OpenCL-Docs/pull/263/commits/adeb8aae4ab18510f35a84cb6db53c4bbd716d32